pkg/logs/tailers/windowsevent/transform.go

```diff
@@ -0,0 +1,208 @@
+func normalizeEventID(mv mxj.Map) error {
+	values, err := mv.ValuesForPath(eventIDPath)
+	if err != nil || len(values) == 0 {
+		return fmt.Errorf("could not find path: %s", eventIDPath)
+	}
+	for _, value := range values {
+		valueMap, ok := value.(map[string]interface{})
+		if !ok {
+			continue
+		}
+
+		// Get element value
+		text, foundText := valueMap["#text"]
+		// Qualifier optional
+		qualifier, foundQualifier := valueMap["Qualifiers"]
+		if foundText && foundQualifier {
+			// Remove Qualifiers attribute from EventID by
+			// overwriting the path with just the text value
+			_ = mv.SetValueForPath(text, eventIDPath)
+			// Add qualifier value to a new path
+			_ = mv.SetValueForPath(qualifier, eventIDQualifierPath)
+		}
+	}
+	return nil
+}
+
+// formatEventDataField transforms the fields parsed from <Data Name='NAME1'>VALUE1</Data><Data Name='NAME2'>VALUE2</Data> to
+// a map that will be JSON serialized to {"NAME1": "VALUE1", "NAME2": "VALUE2"}
+// The Name attribute is optional in the Data schema, so the transform may not apply to all events.
+// https://learn.microsoft.com/en-us/windows/win32/wes/eventschema-datafieldtype-complextype
+func formatEventDataField(mv mxj.Map) error {
+	values, err := mv.ValuesForPath(dataPath)
+	if err != nil || len(values) == 0 {
+		// Event.EventData.Data is an optional element, so it not existing is not an error
+		return nil
+	}
+
+	nameTextMap := make(map[string]interface{})
+	for _, value := range values {
+		valueMap, ok := value.(map[string]interface{})
+		if !ok {
+			continue
+		}
+		name, foundName := valueMap["Name"]
+		text, foundText := valueMap["#text"]
+		if !foundName || !foundText {
+			continue
+		}
+		nameString, ok := name.(string)
+		if !ok {
+			continue
+		}
+		nameTextMap[nameString] = text
+	}
+
+	if len(nameTextMap) > 0 {
+		err = mv.SetValueForPath(nameTextMap, dataPath)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// formatEventBinaryData formats the field Event.EventData.Binary field.
+// The field is optional, so it may not exist in all events.
+// If the field exists, its value is a hex string of arbitrary data.
+// If the hex string contains a utf-16 string, this function will decode it.
+// https://learn.microsoft.com/en-us/windows/win32/wes/eventschema-eventdatatype-complextype
+func formatEventBinaryData(mv mxj.Map) error {
+	values, err := mv.ValuesForPath(binaryPath)
+	if err != nil || len(values) == 0 {
+		// Event.EventData.Data is an optional element, so it not existing is not an error
+		return nil
+	}
+
+	valueString, ok := values[0].(string)
+	if !ok {
+		return fmt.Errorf("could not cast binary data to string: %s", err)
+	}
+
+	// decoded is an utf-16 array of byte
+	decodedHex, err := hex.DecodeString(valueString)
+	if err != nil {
+		return err
+	}
+
+	// TODO: compat: binary data is not guaranteed to be a utf-16 string, but go's
```

Comment:
Is this something that you wanted to address in this PR?

Reply:
Not in this PR, no. It's not ideal but keeping it for now to maintain backwards compatibility. I think we'll have to add an option to control it.